### ISSUE TITLE
Forms not working in deno with dev server

### ISSUE THREAD

**Issue by cotyhamilton** [honojs/honox#229]
### What version of HonoX are you using?

0.1.25

### What steps can reproduce the bug?

Requires Deno 2

```bash
deno upgrade
deno upgrade rc
```

Repro: https://github.com/cotyhamilton/deno-honox

```bash
deno install
deno task dev
```

Vist http://localhost:5173/form and submit form

Additionally you can follow the instructions in the README and create form route to reproduce manually

### What is the expected behavior?

The form is submitted successfully

### What do you see instead?

The server errors with the message `Missing content type` in console and Internal Server Error 500 response returned to the client

### Additional information

This only happens when running the dev server. If you build and serve the output the form submits successfully

```bash
deno task build # deno run -A npm:vite build
deno task preview # (cd dist && deno serve -R index.js)
```

**Comment by yusukebe** [comment]
Hi @cotyhamilton 

Is this issue still occurring?

**Comment by cotyhamilton** [comment]
Yes it does. I just updated the deps in the linked repo and tested.

Or an alternate way is just scaffold a honox project with the cli and replace the adapter with `@hono/vite-dev-server/node` and create a form route

This is with latest deno version 2.1.4. (it could be an issue in deno instead of honox/hono vite dev server).

Thanks for looking!

**Comment by yusukebe** [comment]
@cotyhamilton 

This is not a bug of HonoX. It may be a Deno-side bug when you use `vite` and `@hono/vite-dev-server`.

Repro: https://github.com/yusukebe/deno-form-with-vite-dev-server

With the above repo, the app can't get the header values from the form request. Run the `deno task dev` and request `POST /`, the header value is blank:

![CleanShot 2025-01-01 at 14 23 05@2x](https://github.com/user-attachments/assets/2cc918a0-d59e-4c25-b053-8f2cfb861cc4)

But, you can get the values correctly when run the `vite` with `npm run dev`:

![CleanShot 2025-01-01 at 14 23 27@2x](https://github.com/user-attachments/assets/7ac8c22d-9b4d-4fce-abff-e91595f998f7)

So, this means it's not HonoX matter, and the behavior is expected in a Node.js environment. I think it's also not a bug of `@hono/vite-dev-server`. We don't need to fix it.

**Comment by cotyhamilton** [comment]
Oh yeah! Thanks, I remember reading about this now, I'll have to try and track down that issue
